dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -418,6 +424,27 @@ private void bulkUpsertWorkflows(User authUser) {
         }
     }
 
+    private void checkToolTokens(User authUser, Long userId, String organization) {
+        List<Token> tokens = tokenDAO.findByUserId(userId);
+        List<Tool> tools = userContainers(authUser, userId);
+        if (organization != null && !organization.isEmpty()) {
+            tools.removeIf(tool -> !tool.getNamespace().equals(organization));
+        }
+        Token gitLabToken = Helper.extractToken(tokens, TokenType.GITLAB_COM.toString());
+        Token quayioToken = Helper.extractToken(tokens, TokenType.QUAY_IO.toString());
+        Set<Registry> uniqueRegistry = new HashSet<>();
+        tools.forEach(tool -> uniqueRegistry.add(tool.getRegistry()));
+        if (uniqueRegistry.size() == 0 && quayioToken == null) {
+            throw new CustomWebApplicationException("Please add a Quay.io token", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Elaborate

Reply:
Thought it was a agreed upon that when the user has no tools and no Quay.io token, the refresh all should notify the user that they have no token rather than than successfully completing with no changes.

Reply:
As in elaborate in the error message